### ISSUE TITLE
ENH: `get_prediction` inferential statistics for predict_prob, delta method, count models

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#7601]
The new count models have many `which`, prediction of nonlinear functions of parameters.
But we do not have `get_prediction` for them.

Getting standard errors and confidence intervals requires delta method. In many cases, the prediction functions cannot be reduced to a univariate function and end-point transformation does not apply.

#3415 predict_cdf, predict_prob for discrete/GLM
#2296 PR for delta method and nonlinear wald test

**Comment by josef-pkt** [comment]
another idea

Using delta method does not impose domain restrictions, e.g. probabilities in [0, 1]. 
The common solution is to compute confidence interval in a transformed parameter space. We do this with link functions in one-parameter (linpred) cases.

Can we do an analogue in the multi-parameter delta-method case?
idea is to apply delta-method to predict a nonlinear transformation of the target statistic, e.g. logit or fisher transformd probabilities, and then use endpoint transformation for the actual predicted statistic.

example case would be discrete NegativeBinomial with extra parameter for dispersion.

```
prob = f(mu, alpha)
prob_transformed = g^{-1}(f(mu, alpha))  ~ N(a, s2)
prob = g(prob_transformed)
ci_prob = g(ci_prob_transformed)
```

where g is inverse_link

**Comment by bashtage** [comment]
For scalar confidence intervals, if Pr(a<X<b) = 1-alpha, then Pr(f(a)<f(X)<f(b)) = 1-alpha for monotonic f.  The only down side is that these CIs are no longer symmetric.

**Comment by josef-pkt** [comment]
Yes, that's what prediction with link functions currently does, and what we and Stata do for eform exp(beta) of parameters.
Having asymmetric CI is an advantage if the symmetric CI would have e.g. probabilities >1 or < 0.

The problem here is that the scalar predicted value depends on more than one parameter e.g. linear predictor and dispersion parameter in NB and generalized Poisson or two linear predictors in zero-inflated models and beta regression.

(aside from this, not all predicted statistics are globally monotonic in the parameters.)

endpoint transformation CI could be extended to two parameter case #7182

**Comment by josef-pkt** [comment]
related: comparing two predictions, either two "obs" of exog, or average over two arrays (many obs) of exog can be nonlinear prediction (in link models)
#2291

e.g. risk ratio,  or risk diff or margin for unit change in exog #2215
e.g. extending proportion_2indep analysis to regression models

need confidence interval and inference for those.
